File: solutions/1725/main.py
import re
from typing import List 

class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        max_len = -float("inf") 
        result = 0 
        for r in rectangles: 
            if max_len < min(r[0], r[1]): 
                result = 1 
                max_len = min(r[0], r[1])
            elif max_len == min(r[0], r[1]): 
                result += 1 
        return result 

if __name__ == "__main__": 
    rectangles = [[5,8],[3,9],[5,12],[16,5]]
    s = Solution()
    assert s.countGoodRectangles(rectangles) == 3 

File: solutions/1725/main.rs
struct Solution {}

impl Solution {
    pub fn count_good_rectangles(rectangles: Vec<Vec<i32>>) -> i32 {
        let mut max_len: i32 = i32::MIN; 
        let mut result: i32 = 0; 
        let mut min_side: i32 = 0; 
        rectangles.iter().for_each(|r| {
            min_side = r[0].min(r[1]); 
            if max_len < min_side {
                max_len = min_side; 
                result = 1; 
            } 
            else if max_len == min_side {
                result += 1; 
            }
        });
        return result; 
    }
}

fn main() {
    assert_eq!(Solution::count_good_rectangles(vec![vec![5,8],vec![3,9],vec![5,12],vec![16,5]]), 3); 
}

